wgpu-hal/src/gles/egl.rs

```diff
@@ -606,7 +605,29 @@ impl Inner {
             context_attributes.push(khr_context_flags);
         }
         context_attributes.push(khronos_egl::NONE);
-        let context = match egl.create_context(display, config, None, &context_attributes) {
+
+        gl_context_attributes.extend(&context_attributes);
+        gles_context_attributes.extend(&context_attributes);
+
+        let context = match egl
```

Comment:
I had a pretty hard time parsing this with all the functionality crammed inside the head of a match statement. Also the error message down there isn't entirely correct anymore. How about

```rust
        let context = if supports_opengl {
            if let Ok(context) = egl.create_context(display, config, None, &gl_context_attributes) {
                context
            } else {
                // We tried OpenGL earlier, but failed. Try OpenGL ES instead.
                egl.bind_api(khronos_egl::OPENGL_ES_API).unwrap();
                egl.create_context(display, config, None, &gles_context_attributes)
                    .map_err(|err| {
                        crate::InstanceError::with_source(
                            String::from("unable to create either OpenGL and GLES3.x context"),
                            err,
                        )
                    })?
            }
        } else {
            egl.create_context(display, config, None, &gles_context_attributes)
                .map_err(|err| {
                    crate::InstanceError::with_source(
                        String::from("unable to create GLES 3.x context"),
                        err,
                    )
                })?
        };
```

Reply:
I've refactored it a bit to be a bit clearer.

Reply:
ah neat, even better than what I suggested!